aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -283,33 +285,17 @@ impl<T: Transaction> CapturedReads<T> {
         &mut self,
         id: T::Identifier,
         value: DelayedFieldValue,
-    ) -> anyhow::Result<()> {
-        let ret = match self.delayed_field_reads.entry(id) {
+    ) -> anyhow::Result<(), PanicError> {
+        match self.delayed_field_reads.entry(id) {
             Vacant(e) => {
                 e.insert(value);
-                UpdateResult::Inserted
-            },
-            Occupied(mut e) => {
-                let existing_read = e.get_mut();
-                if value == *existing_read {
-                    UpdateResult::Updated
-                } else {
-                    UpdateResult::Inconsistency(format!(
-                        "Read {:?} must be consistent with the already stored read {:?}",
-                        value, existing_read
-                    ))
-                }
+                // Inserted
+                Ok(())
             },
-        };
-
-        match ret {
-            UpdateResult::IncorrectUse(_) => unreachable!(),
-            UpdateResult::Inconsistency(m) => {
-                // Record speculative failure.
-                self.speculative_failure = true;
-                bail!(m);
-            },
-            UpdateResult::Updated | UpdateResult::Inserted => Ok(()),
+            Occupied(e) => Err(code_invariant_error(format!(
```

Comment:
Because we cache in extension, right?

Reply:
yes. with new change, this will change, though.